src/core-services/account/mutations/removeUserPermissions.js

```diff
@@ -0,0 +1,74 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+
+const inputSchema = new SimpleSchema({
+  "accountId": String,
+  "groups": {
+    type: Array, // groupIds that user belongs to
+    minCount: 1
+  },
+  "groups.$": {
+    type: String
+  }
+});
+
+/**
+ * @name accounts/removeUserPermissions
+ * @memberof Mutations/Accounts
+ * @summary Removes a group from an account (user) group. This addition to an account  group effectively
+ * removes permissions from account (user)
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {Object} input.groups - groups to append to
+ * @param {String} input.accountId - decoded ID of account on which entry should be updated
+ * @returns {Promise<Object>} with updated account
+ */
+export default async function removeUserPermissions(context, input) {
+  const itemsToValidate = { accountId: input.accountId, groups: input.groups };
+  inputSchema.validate(itemsToValidate);
+  const { appEvents, collections, userId: userIdFromContext } = context;
+  const { Accounts } = collections;
+  const { groups, shopId, accountId } = input;
+
+
+  const account = await Accounts.findOne({ _id: accountId });
+
+  if (!account) throw new ReactionError("not-found", "No account found");
+
+  if (!context.isInternalCall) {
+    await context.validatePermissions("reaction:accounts", "update", { shopId, legacyRoles: ["reaction-accounts"] });
+  }
+
+
+  // Update the Reaction Accounts collection with new groups info
+  const { value: updatedAccount } = await Accounts.findOneAndUpdate(
+    {
+      _id: accountId
+    },
+    {
+      $pull: {
+        groups: {
+          $in: groups
```

Comment:
This removes the user from the group, but doesn't remove the `roles` of that group from the `user` record in the `users` collection. Some checks still seem to rely on roles from there.  At least that's what it looks like from hasPermissions function `https://github.com/reactioncommerce/reaction/blob/release-3.0.0/src/plugins/legacy-authorization/util/hasPermission.js`